solutions/usaco-1107.mdx

```diff
@@ -66,5 +66,83 @@ print(abs(ans))
 ```
 
 </PySection>
+	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class YearOfTheCow {
+	static String[] animals = {"Ox", "Tiger", "Rabbit", "Dragon", "Snake", 
+	"Horse", "Goat", "Monkey", "Rooster", "Dog", "Pig", "Rat"};
+
+	static Map<String, Integer> when_born = new TreeMap<>();
+
+	// given a year, return what animal the year is
+	static String getAnimal(int year)
+	{
+		int a = 0, y = 2021;
+		while (y < year) {
+			y++;
+			a++;
+			if (a == 12) {  // reset once you reach 12
```

Comment:
Maybe something a little more descriptive/specific - end of 12-year cycle or something like that